expansion/src/main/java/no/unit/nva/expansion/model/ExpandedResource.java

```diff
@@ -180,12 +187,45 @@ public String toString() {
         return toJsonString();
     }
 
-    private static JsonNode addFields(String json, Publication publication) {
-        var sortedJson = strToJsonWithSortedContributors(json);
-        injectHasFileEnum(publication, (ObjectNode) sortedJson);
-        expandLicenses(sortedJson);
-        injectJoinField(publication, (ObjectNode) sortedJson);
-        return sortedJson;
+    private static ObjectNode addFields(String jsonString, Publication publication) {
+        var objectNode = strToJson(jsonString);
+        sortContributors(objectNode);
+        injectHasFileEnum(publication, objectNode);
+        expandLicenses(objectNode);
+        injectJoinField(publication, objectNode);
+        injectContributorCount(objectNode);
+        injectContributorsPreview(objectNode);
+        return objectNode;
+    }
+
+    private static void injectContributorCount(ObjectNode json) {
+        var contributors = json.at(CONTRIBUTORS_PTR);
+        if (!contributors.isMissingNode() && contributors.isArray()) {
+            var entityDescription = (ObjectNode) json.at(ENTITY_DESCRIPTION_PTR);
+            if (!entityDescription.isMissingNode() && entityDescription.isObject()) {
+                entityDescription.put(CONTRIBUTORS_COUNT, contributors.size());
+            }
+        }
+    }
+
+    private static void injectContributorsPreview(ObjectNode json) {
+        var contributors = json.at(CONTRIBUTORS_PTR);
+        if (!contributors.isMissingNode() && contributors.isArray()) {
+            var entityDescription = (ObjectNode) json.at(ENTITY_DESCRIPTION_PTR);
+            if (!entityDescription.isMissingNode() && entityDescription.isObject()) {
+                List<JsonNode> contributorsList = new ArrayList<>();
+                contributors.forEach(contributorsList::add);
+                var sortedContributors = contributorsList.stream()
+                                             .sorted(Comparator.comparingInt(c -> c.get(CONTRIBUTOR_SEQUENCE).asInt()))
+                                             .sorted(ExpandedResource::sortContributorByVerifiedFirst)
+                                             .limit(MAX_CONTRIBUTORS_PREVIEW).toList();
+
+                ArrayNode arrayNode = new ArrayNode(JsonNodeFactory.instance);
+                arrayNode.addAll(sortedContributors);
+
+                entityDescription.put(CONTRIBUTORS_PREVIEW, arrayNode);
```

Comment:
`put` is deprecated, use `entityDescription.set(CONTRIBUTORS_PREVIEW, arrayNode);`